File: flowai-backend/app/schemas/address.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict, Field


class AddressBase(BaseModel):
    label: str | None = Field(default=None, max_length=60)
    full_name: str = Field(..., min_length=1, max_length=255)
    phone: str | None = Field(default=None, max_length=50)
    line1: str = Field(..., min_length=1, max_length=200)
    line2: str | None = Field(default=None, max_length=200)
    city: str = Field(..., min_length=1, max_length=120)
    state: str | None = Field(default=None, max_length=120)
    postal_code: str | None = Field(default=None, max_length=40)
    country: str = Field(default="US", min_length=2, max_length=2)
    is_default: bool = False


class AddressCreate(AddressBase):
    pass


class AddressUpdate(BaseModel):
    label: str | None = Field(default=None, max_length=60)
    full_name: str | None = Field(default=None, min_length=1, max_length=255)
    phone: str | None = Field(default=None, max_length=50)
    line1: str | None = Field(default=None, min_length=1, max_length=200)
    line2: str | None = Field(default=None, max_length=200)
    city: str | None = Field(default=None, min_length=1, max_length=120)
    state: str | None = Field(default=None, max_length=120)
    postal_code: str | None = Field(default=None, max_length=40)
    country: str | None = Field(default=None, min_length=2, max_length=2)
    is_default: bool | None = None


class AddressRead(AddressBase):
    id: int
    created_at: datetime
    updated_at: datetime

    model_config = ConfigDict(from_attributes=True)

File: flowai-website-flontend/src/lib/validators/address.ts
import { z } from "zod";

export const addressSchema = z.object({
  label: z.string().max(60).optional().or(z.literal("")),
  full_name: z.string().min(2, "Enter the recipient name"),
  phone: z.string().max(50).optional().or(z.literal("")),
  line1: z.string().min(2, "Enter a street address"),
  line2: z.string().max(200).optional().or(z.literal("")),
  city: z.string().min(2, "Enter a city"),
  state: z.string().max(120).optional().or(z.literal("")),
  postal_code: z.string().max(40).optional().or(z.literal("")),
  country: z.string().length(2, "Use a 2-letter country code"),
  is_default: z.boolean().optional(),
});

export type AddressInput = z.infer<typeof addressSchema>;
